sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCWriteSuite.scala

```diff
@@ -44,9 +44,6 @@ class JDBCWriteSuite extends SharedSQLContext with BeforeAndAfter {
 
   val testH2Dialect = new JdbcDialect {
     override def canHandle(url: String) : Boolean = url.startsWith("jdbc:h2")
-    override def getCatalystType(
```

Comment:
Thank you for review, @gatorsmile .

`JdbcDialect` should handle a few DBMS specific types and should return `None` for general types. The schema is handled in general in the following fallback logic.

https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala#L245-L246

The current implementation is wrong because it returns always `StringType` for all kind types of columns.